### PR TITLE
OHRM5X-1657: Develop dashboard - my actions summary widget

### PR DESCRIPTION
**Pull request by ManulMax**

### CODE REVIEW

**Comment by Super-Chama** [comment]
@ManulMax Please update oxd version to `1.0.7-alpha.5` after https://github.com/orangehrm/oxd/pull/601 is merged. you can use icons form there.

**Review by RajithaKumara — APPROVED** [review]

File: src/client/src/orangehrmAttendancePlugin/pages/ViewEmployeeAttendanceSummary.vue [hunk 1]
```diff
@@ -133,7 +133,7 @@ export default {
       return data.map(item => {
         return {
           id: item.empNumber,
-          empName: $tEmpName(this.employee, {
+          empName: $tEmpName(item, {
```
Comment (Super-Chama) [inline src/client/src/orangehrmAttendancePlugin/pages/ViewEmployeeAttendanceSummary.vue:136]: Have you done developer testing on this? confirmed working?

File: src/client/src/orangehrmDashboardPlugin/components/MyActionSummeryView.vue [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    :empty="isEmpty"
+    :empty-text="emptyText"
+    :loading="isLoading"
+    :title="$t('general.my_actions')"
+  >
+    <div v-if="myActions.length > 0" class="orangehrm-todo-list">
+      <div v-if="leaveRequestCount > 0" class="orangehrm-todo-list-item">
+        <oxd-icon-button
+          class="orangehrm-report-icon"
+          name="clock"
```
Comment (Super-Chama) [inline src/client/src/orangehrmDashboardPlugin/components/MyActionSummeryView.vue:33]: please update the icons

File: src/client/src/orangehrmDashboardPlugin/components/MyActionSummeryView.vue [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    </div>
+  </base-widget>
+</template>
+<script>
+import {navigate} from '@ohrm/core/util/helper/navigation';
+import {APIService} from '@/core/util/services/api.service';
+import BaseWidget from '@/orangehrmDashboardPlugin/components/BaseWidget.vue';
+
+export default {
+  name: 'MyActionSummeryViewWidget',
```
Comment (Super-Chama) [inline src/client/src/orangehrmDashboardPlugin/components/MyActionSummeryView.vue:99]:
```suggestion
  name: 'MyActionSummaryWidget',
```
typo ~~Summery~~  ==> summary

File: src/client/src/orangehrmDashboardPlugin/components/MyActionSummeryView.vue [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+        this.isLoading = false;
+      });
+  },
+
+  methods: {
+    onClickPendingLeave() {
+      navigate('/leave/viewLeaveList');
+    },
+    onClickPendingTimesheet() {
+      navigate('/time/viewEmployeeTimesheet');
```
Comment (Super-Chama) [inline src/client/src/orangehrmDashboardPlugin/components/MyActionSummeryView.vue:167]: hope these pages get loaded with current data

File: src/client/src/orangehrmDashboardPlugin/components/MyActionSummeryView.vue [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+      navigate('/performance/searchEvaluatePerformanceReview');
+    },
+    onClickPendingInterview() {
+      navigate('/recruitment/viewCandidates');
+    },
+  },
+};
+</script>
+<style lang="scss" scoped>
+@import '@ohrm/oxd/styles/_mixins.scss';
```
Comment (Super-Chama) [inline src/client/src/orangehrmDashboardPlugin/components/MyActionSummeryView.vue:179]: move to stylesheet please

File: src/client/src/orangehrmDashboardPlugin/pages/ViewDashboard.vue [hunk 6]
```diff
@@ -26,6 +26,12 @@
     >
       <employee-attendance-widget></employee-attendance-widget>
     </oxd-grid-item>
+    <oxd-grid-item
+      v-if="$can.read('admin_widgets')"
```
Comment (Super-Chama) [inline src/client/src/orangehrmDashboardPlugin/pages/ViewDashboard.vue:30]: this is wrong i think. my action summary widget is available for everyone

File: src/client/src/orangehrmDashboardPlugin/pages/ViewDashboard.vue [hunk 7]
```diff
@@ -52,13 +58,15 @@ import EmployeeSubunitWidget from '@/orangehrmDashboardPlugin/components/Employe
...
 import EmployeeAttendanceWidget from '@/orangehrmDashboardPlugin/components/EmployeeAttendanceWidget.vue';
+import MyActionSummeryViewWidget from '@/orangehrmDashboardPlugin/components/MyActionSummeryView.vue';
 
 export default {
   components: {
     'employee-subunit-widget': EmployeeSubunitWidget,
     'employee-location-widget': EmployeeLocationWidget,
     'employees-on-leave-widget': EmployeesOnLeaveWidget,
     'employee-attendance-widget': EmployeeAttendanceWidget,
+    'myaction-summery-view-widget': MyActionSummeryViewWidget,
```
Comment (Super-Chama) [inline src/client/src/orangehrmDashboardPlugin/pages/ViewDashboard.vue:69]: correct typo summary

File: src/client/src/orangehrmLeavePlugin/pages/LeaveRequest.vue [hunk 8]
```diff
@@ -111,6 +111,7 @@ import usei18n from '@/core/util/composable/usei18n';
 import useDateFormat from '@/core/util/composable/useDateFormat';
 import {formatDate, parseDate} from '@/core/util/helper/datefns';
 import useLocale from '@/core/util/composable/useLocale';
+import useEmployeeNameTranslate from '@/core/util/composable/useEmployeeNameTranslate';
```
Comment (Super-Chama) [inline src/client/src/orangehrmLeavePlugin/pages/LeaveRequest.vue:114]: please dev test this screen and confirm working

**Review by Super-Chama — CHANGES_REQUESTED** [review]

File: src/client/src/orangehrmDashboardPlugin/components/MyActionSummeryView.vue (hunk 3 above)
Comment (RajithaKumara) [reply src/client/src/orangehrmDashboardPlugin/components/MyActionSummeryView.vue:99]: nice catch. @ManulMax please refactor rename the file name also

File: src/client/src/orangehrmDashboardPlugin/pages/ViewDashboard.vue [hunk 9]
```diff
@@ -26,6 +26,12 @@
     >
       <employee-attendance-widget></employee-attendance-widget>
     </oxd-grid-item>
+    <oxd-grid-item
+      v-if="$can.read('admin_widgets')"
+      class="orangehrm-dashboard-widget"
+    >
+      <myaction-summery-view-widget></myaction-summery-view-widget>
```
Comment (RajithaKumara) [inline src/client/src/orangehrmDashboardPlugin/pages/ViewDashboard.vue:33]:
```suggestion
      <myaction-summary-view-widget></myaction-summary-view-widget>
```

File: src/client/src/orangehrmLeavePlugin/pages/LeaveRequest.vue [hunk 10]
```diff
@@ -211,6 +213,7 @@ export default {
       processLeaveAction,
       jsDateFormat,
       locale,
+      translateEmpName: $tEmpName,
```
Comment (RajithaKumara) [inline src/client/src/orangehrmLeavePlugin/pages/LeaveRequest.vue:216]: basic mistakes. should be catch when dev testing

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

**Review by Super-Chama — APPROVED** [review]

File: src/client/src/orangehrmAttendancePlugin/pages/ViewEmployeeAttendanceSummary.vue (hunk 1 above)
Comment (ManulMax) [reply src/client/src/orangehrmAttendancePlugin/pages/ViewEmployeeAttendanceSummary.vue:136]: Working

**Review by RajithaKumara — APPROVED** [review]

File: src/client/src/orangehrmDashboardPlugin/components/MyActionSummeryView.vue (hunk 4 above)
Comment (ManulMax) [reply src/client/src/orangehrmDashboardPlugin/components/MyActionSummeryView.vue]: Working

File: src/client/src/orangehrmLeavePlugin/pages/LeaveRequest.vue (hunk 8 above)
Comment (ManulMax) [reply src/client/src/orangehrmLeavePlugin/pages/LeaveRequest.vue:114]: Working

**Pull request merged**